### ISSUE TITLE
Provisioning Profile "com.company.app AppStore" doesn't include signing certificate iPhone...

### ISSUE THREAD

**Issue by meehanman** [fastlane/fastlane#13079]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject

I'm having codesigning issues when using gym for an iOS app; in particular the codesigning_identity (Signing Certificate) is not being set by the build_app command.
 
#### Question Description

I expect cert and sigh to check certificates and provisioning profiles and then be able to set these profiles in my XCode project using build_app (gym), but the following Factfile returns the error. `Code Signing Error: Provisioning profile "com.CompanyName.myCompanyName AppStore" doesn't include signing certificate "iPhone Distribution: Company Name (XZXZXZXZXZX)".`

	default_platform(:ios)

	platform :ios do
	  desc "Push a new beta build to TestFlight"
	  lane :beta do |options|
		  build_number = increment_build_number(
			  build_number: latest_testflight_build_number + 1
		  )
		  version_number = increment_version_number(
			version_number: options[:version_number]
		  )
		  automatic_code_signing(use_automatic_signing: false, team_id: "XZXZXZXZXZX")

		  sh(
			  "echo ##teamcity[buildNumber '#{version_number}(#{build_number})']"
		  )
		  cert
		  uuid = sigh(
			force: true,
			filename: "CompanyName.mobileprovision"
		  )
		  update_project_provisioning(
			target_filter: "CompanyName",
			profile: "CompanyName.mobileprovision",
			build_configuration: options[:configuration]
		  )
		  
		  build_app(
			  scheme: "CompanyName",
			  configuration: options[:configuration],
			  clean: true,
			  silent: true,
			  output_directory: "/usr/local/Repos/OLS/artifacts/",
			  output_name: "CompanyName.ipa",
			  export_method: "app-store",
			  codesigning_identity: "iPhone Distribution: Company Name (XZXZXZXZXZX)",
			  export_options: {
				  provisioningProfiles: {
					  "com.CompanyName.myCompanyName" => "com.CompanyName.myCompanyName AppStore",
					  "com.CompanyName.myCompanyName" => "com.CompanyName.myCompanyName Development"
				  }
			  }
		  )

		  if options[:pushToStore]
			  upload_to_testflight(
				  ipa: "/usr/local/Repos/OLS/artifacts/CompanyName.ipa",
			  )
		  end
	  end
	end

After running fastlane and recieving the error. I'm left with an XCode project that states:
`Code Signing Error: Provisioning profile "com.CompanyName.myCompanyName AppStore" doesn't include signing certificate "iPhone `**Developer**`: Company Name (`**XYXYXYXYXYX**`)".`

----

If I manually then reselect the Provisioing Profile for the "Signing (Release)" back to the same one that was set. The Signing Certificate then changes to the one specified **iPhone Distribution: Company Name (XZXZXZXZXZX)**

I'm aware this is a code signing issue, and maybe not a Fastlane issue, but any help is appriciated to narrow down this issue.

Thank you
Dean Meehan

**Comment by LasseSLambertsen** [comment]
Got the same problem, I can sign fine using Xcode directly, but not through fastlane

**Comment by meehanman** [comment]
I actually got this sorted. Although it's possibly not a fault with XCODE and I'd love to sit down and write a detailed guide /procedure to fix all coffee signing issues.

I fixed it by:
1. Backing up all your certificates by exporting them
2. Delete all your iphone certificates
3. Try and use cert and sigh to generate them.
4. Restart the Mac
5. Re-add your certs 1 by 1 while trying build_app again and again.

Try them steps to see how far you get. If you get an error fffffffffffff, I just reverted from svn to a working status.

Good luck

On 11 Aug 2018 20:31, LasseSLambertsen <notifications@github.com> wrote:

Got the same problem, I can sign fine using Xcode directly, but not through fastlane
